homeassistant/components/opensprinkler/__init__.py

```diff
@@ -0,0 +1,146 @@
+"""Opensprinkler integration."""
+import asyncio
+import logging
+
+from pyopensprinkler import OpenSprinkler
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_PORT, STATE_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv, discovery
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.util import Throttle
+
+from .const import (
+    CONF_RUN_SECONDS,
+    DATA_DEVICES,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DEFAULT_RUN_SECONDS,
+    DOMAIN,
+    SCAN_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_DEVICES = "devices"
+SUPPORTED_DOMAINS = ["binary_sensor", "scene", "sensor", "switch"]
+
+DEVICE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_RUN_SECONDS, default=DEFAULT_RUN_SECONDS): cv.positive_int,
+    }
+)
+
+CONFIG_SCHEMA = vol.Schema(
```

Comment:
Please use a config flow to set up the integration. You can generate one with `python3 -m script.scaffold config_flow`

Reply:
Please remove the YAML configuration. This is no longer supported.